### ISSUE TITLE
Chase Bank UK

### ISSUE THREAD

**Issue by ojoeth** [akc3n/banking#69]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

Chase Bank UK

### Link to app

https://play.google.com/store/apps/details?id=com.chase.intl

### App version

1.10.1

### Country of the app

United Kingdom

### Build Number

SP2A.220405.004.2022041100

### Device

- [ ] Pixel 6 Pro
- [X] Pixel 6
- [ ] Pixel 5a
- [ ] Pixel 5
- [ ] Pixel 4a (5G)
- [ ] Pixel 4a
- [ ] Pixel 4 XL
- [ ] Pixel 4
- [ ] Pixel 3a XL
- [ ] Pixel 3a
- [ ] Pixel 3 XL
- [ ] Pixel 3

### Google Play installed

- [X] Installed
- [ ] Not installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### SafetyNet Enforcement

- [ ] Enforced
- [ ] Not enforced
- [X] Unsure

### Native code debugging

- [ ] Enabled
- [X] Disabled

### Stock OS compatibility

- [ ] Works
- [ ] Does not work
- [X] Not tested

### Profile app tested in

- [X] Main user profile
- [ ] Secondary user profile
- [ ] I used a work profile device manager app (Please mention more details about it down below like the app)

### Description of the app's functionality

Everything works, except for Home -> Rewards -> How Rewards Work. This seems to give an error (https://i.imgur.com/IGaa4KN.png)

**Comment by akc3n** [comment]
@ojoeth hey there! Thanks so much! I'll get that added shortly.

**Comment by ojoeth** [comment]
Adding to this, yesterday I got a message which told me the app would not run because I have a "rooted/jailbroken device", however I was able to dismiss the message and continue using the app as normal. I have not had any other of these messages since I got that one yesterday.

**Comment by break6700** [comment]
@ojoeth Hi! just checking if this is still correct info as it's been a year and I believe Chase UK now uses Google's Play Integrity API

**Comment by eylenburg** [comment]
The Chase app still works as of 24 Nov 2023. No issues at all. No Play Services needed.

**Comment by akc3n** [comment]
@eylenburg thank you for the status update!

**Comment by LouisJackman** [comment]
After a couple of months of using Chase on GrapheneOS successfully (Pixel 6 Pro), issues started occurring at roughly the same time in the last few weeks. Specifically, transferring Rewards and certain in-app approval prompts never complete after pressing "Approve" or "Transfer", e.g.:

- Approval for 3DS online payments.
- Approval for adding to a Pixel Watch wallet (note that Wallet for payment cards on the phone itself doesn't work across the board as stated elsewhere).
- Approval during a customer service call.

**It's unclear if this is a recent regression against GrapheneOS in general or specific to my device. It'd be great to hear from other Chase users on GrapheneOS if they can corroborate.**

Miscellaneous info:

- App data wipe and reinstall attempted, but didn't address the issue.
- Tried with both " Exploit protection compatibility mode" on and off.
- Still functions fine as a chip-and-PIN and contactless card account and can do non-Rewards transfers. Just not the above tasks.

**Comment by HaymanTam** [comment]
Long time user with Chase on Pixel 6.  I can concur with issue @LouisJackman mentioned. 

Anything that requires additional pass approval doesn't work right now. 
The App will just stuck in infinite loading screen after entering your own passcode.  This includes:
- Approval for online payee
- Adding new payee

What I have tried: 
Reinstalling and Forcestop app
Turning on "Exploit protection compatibility mode" 

Not familiar with GrapaheneOS development so couldn't comment more.

**Comment by portalbyte** [comment]
I have the same issues as above (Pixel 7 GrapheneOS).

HOWEVER, I also had this issue previously on LineageOS, so this isn't a GrapheneOS issue (although LineageOS now no longer works as Chase requires a locked bootloader).

**Comment by Bevetkia** [comment]
> After a lot of back and forth with Chase, they are saying they've solved the issue in the next app update (v2.35) and will be rolled out between 12th and 17th June. I'll update it as soon as it's available and report back. Fingers crossed it will work this time.

I also got the same confirmation today. That the next update should fix this issue.

**Comment by portalbyte** [comment]
@LouisJackman @HaymanTam @nonalloc @Bevetkia I have just recieved the update to v2.35.0. Just tried adding a new payee and it works fine - no issues entering my passcode. I assume it has been fixed.

**Comment by Permanently** [comment]
If you're currently getting crashes on Chase, make sure to disable **Hardened memory allocator**

**Comment by vasubhatia-acn** [comment]
Getting crashes now. Unable to login anymore. Hardened malloc disabled.

**Comment by Permanently** [comment]
Latest update still working for me, with these settings - see how this goes?

![Screenshot_20260504-125558~2.png](https://github.com/user-attachments/assets/1cadf4dd-67d6-4632-a91b-fc2401f9e0df)

**Comment by vasubhatia-acn** [comment]
I was initially logged in, but got logged out and the app hasn't worked for me since. I guess it's something to do with the play integrity API.

<img width="1080" height="2424" alt="Image" src="https://github.com/user-attachments/assets/2be1d91f-a9bc-48d9-91e5-a0c12b1e01af" />

**Comment by vasubhatia-acn** [comment]
Yeah, it throws an error if you have play services installed. Blocking play API requests still throws an error. Tried it in a work profile without play services and it worked.

**Comment by beaneeber** [comment]
> I was initially logged in, but got logged out and the app hasn't worked for me since. I guess it's something to do with the play integrity API.

Could you please confirm which version you're on? I'm on 2.77.0 and can't replicate the issue.

**Comment by break6700** [comment]
Update on this: As of v2.83.0 (2184) the app is now using the Play Integrity API which results GrapheneOS users being told quote "Your device doesn't meet security standards". This should be re-opened as it seems Chase UK no longer supports GrapheneOS.

**Comment by beaneeber** [comment]
> Update on this: As of v2.83.0 (2184) the app is now using the Play Integrity API which results GrapheneOS users being told quote "Your device doesn't meet security standards". This should be re-opened as it seems Chase UK no longer supports GrapheneOS.

I see the same thing, downgraded to 2.81.0 (2153) then attempted to log in and saw the same error.

As a workaround I blocked Play Integrity API calls and it appears to be working now.

I will contact [ChaseSupportUK@chase.com](mailto:ChaseSupportUK@chase.com) about this issue.

**Comment by SvartrShell** [comment]
Also started experiencing this today - unable to block play integrity API calls from the app menu either - any idea why?

**Comment by epagone** [comment]
I have the same problem as @SvartrShell. I don't have the option to block play integrity API calls for this app (it's available for other banking apps in the same profile) and I don't understand why, as well. I'm using Version 2.83.0 (2184).

Could it be that downgrading to Version 2.81.0 (2153), like @beaneeber mentions, makes the option available again? 🤔

**Comment by cgossa** [comment]
I have the same issue. I did downgraded the app and it worked for a a few hours. Then I got the same error.
Finally contacted them and raised a complaint as I couldn't get this working no matter what I was doing.
I was told that this is raised in both their legal team r garding my concerns of pushing us to use Google and their dev team to figure out why this happens.

**Comment by SvartrShell** [comment]
@epagone @cgossa @beaneeber 

I have this working currently by Having Secure App Spawning Disabled, and Exploit protection Compatibility mode DISABLED, with all the settings below the toggle left on Default.

If I switch Hardened Memory Allocator to Disabled, the app gives the error.

I had to force stop Chase and clear cache after making these tweaks and so far i can get back in 

I have emailed Chase support as well.

Play Integrity API notification hasn't popped up - so I'm not sure if this issue is as a result of Chase implementing the API, or a clash with the GrapheneOS App Spawning and Memory Allocator - not knowledgeable enough to say for sure, so this is purely a theory

**Comment by cgossa** [comment]
@epagone @beaneeber 
Thanks @SvartrShell for this solution.
I can confirm that this worked for me as well. It keeps giving notifications though that it uses the play integrity. Also this only worked in the main profile for me.

**Comment by epagone** [comment]
It works in a secondary profile for me 😉

**Comment by SvartrShell** [comment]
> [@epagone](https://github.com/epagone) [@beaneeber](https://github.com/beaneeber) Thanks [@SvartrShell](https://github.com/SvartrShell) for this solution. I can confirm that this worked for me as well. It keeps giving notifications though that it uses the play integrity. Also this only worked in the main profile for me.

Interesting, does this mean you get the option to disable API calls in the app settings menu? As the option for that doesn't appear for me yet

**Comment by cgossa** [comment]
> > [@epagone](https://github.com/epagone) [@beaneeber](https://github.com/beaneeber) Thanks [@SvartrShell](https://github.com/SvartrShell) for this solution. I can confirm that this worked for me as well. It keeps giving notifications though that it uses the play integrity. Also this only worked in the main profile for me.
> 
> Interesting, does this mean you get the option to disable API calls in the app settings menu? As the option for that doesn't appear for me yet

Yes I can. This option was not there at the first until I got the popup and then it appeared. Which is very weird. I didn't block it though as I don't know if this will kill the app again.

**Comment by beaneeber** [comment]
On reflection, I don't think my suggestion did anything. It just so happened to coincidentally start working... for a while... then I started seeing the same error again.

@SvartrShell @epagone My understanding is that you're only offered this if GOS detects Play Integrity usage, it then notifies you and if you open the notification you have the option to block calls. Not sure if you can do this directly in the app's settings without being prompted.

@cgossa I'm going to follow up on my initial email and file a complaint as well. I suggest everyone else does the same. I contacted them again yesterday via in app chat and they suggested reinstalling the app... so it doesn't seem like they're aware of the scale of the problem or its root cause.

@SvartrShell Thanks for the tip, this worked and I haven't had any issues since.

**Comment by beaneeber** [comment]
Just stumbled upon this thread: https://discuss.grapheneos.org/d/40289-chase-app-error-on-latest-graphene-version/14

**Comment by cgossa** [comment]
@beaneeber I totally agree with you. People should start filing complaints so banks will start realising the extent of people that use gos and have issues with their banks. At the end of the day it is our legal right to protect our personal data and their legal obligation to allow us to. Also filing a complaint in UK triggers a very specific FCA process and if companies do not deal with the complaint in an acceptable way then they end up paying huge fees. So this is the only way really for us to put some pressure on them. So please, if you have issues raise a complaint not just a note.

**Comment by SvartrShell** [comment]
Chase email directed me to their in-app chat (ironic) or a phone call, given that I want written records of this, I have opted for the app chat to raise the complaint. Chat messages have a 1024 character limit the message needs to be sent in blocks - To save some time for others I will paste what I have sent here for others to use if they wish - its aligned to an FCA compliant complaint.

Of course this is entirely personal, just wanted to offer a quick way to raise this as I believe in power in numbers and as @beaneeber  and @cgossa have mentioned already, the only way we will get any movement is to go through official channels


### Messages In Order
```
(1/5) I want to raise a formal complaint, please log this under your complaints process.

Since this week my account access has been blocked, with the app showing an error that my device doesn't meet your security requirements. This happened without any prior notice and with no transition period to arrange alternative access.

This is caused by Chase UK adopting the Play Integrity API in app version 2.83.0 (build 2184). I run GrapheneOS, a hardened, open source version of Android. This is documented on a public banking-app compatibility tracker (github.com/PrivSec-dev/banking-apps-compat-report, issue 69), where other GrapheneOS users report the same "device doesn't meet security standards" message from this same app version.
```

```
(2/5) The only workarounds available to me are: (1) disabling two of GrapheneOS's security hardening features, Secure App Spawning and Exploit Protection Compatibility Mode, both of which reduce the app's exposure to exploitation, or (2) moving to stock Android or iOS, i.e. Google's or Apple's platforms. Neither is acceptable. Option 1 means the app only works when my device is made less secure, which contradicts the stated purpose of the check. Option 2 would force me to adopt platforms I have specifically chosen not to use for privacy and data-control reasons, removing my ability to choose the operating system I run on my own hardware in order to access my own funds.

Chase offers no web-based account access, so app-only enforcement of this leaves affected customers with no alternative route to their own funds.
```

```
(3/5) For reference, GrapheneOS's own documentation on its security model and the technical alternatives to Play Integrity API is publicly available here:
- Security features overview: https://grapheneos.org/features
- Technical guide for developers on supporting hardware attestation as an alternative to Play Integrity API: https://grapheneos.org/articles/attestation-compatibility-guide
- Frequently asked questions on GrapheneOS's security architecture: https://grapheneos.org/faq
- Usage guide, including notes on banking app compatibility: https://grapheneos.org/usage
- Developer/community discussion on Play Integrity workarounds: https://discuss.grapheneos.org/d/23977-workaround-for-play-integrity-on-grapheneos-possible
```

```
(4/5) I'd like written confirmation of:
1. Why the app requires disabling device security hardening to pass a supposed security check
2. Why customers are effectively required to adopt Google or Apple's platforms to retain account access, and whether Chase considers this an acceptable restriction on customer choice
3. Whether Chase's development team has reviewed the hardware attestation alternative documented above, which would allow GrapheneOS devices to be verified without requiring Play Integrity
4. When 2.83.0 was rolled out and why no notice was given
5. Whether Chase will provide an alternative access route for customers on non-Google-certified but secure operating systems
6. Who owns this decision so I can follow up directly

Under DISP 1.6.2R you're required to send a final response within 8 weeks, or explain why not and confirm my right to escalate to the Financial Ombudsman Service. Please also send a written acknowledgement.
```

#### OPTIONAL CLOSING 

```
(5/5) I've held accounts with Chase for many years. If this isn't resolved, or the requirement is kept without a workable alternative, I'll be closing all my accounts.

Please confirm receipt and a complaint reference number.
```

**Comment by break6700** [comment]
UPDATE: GrapheneOS developers have confirmed a fix in the OS to get this working again in the latest code (2026071500). This fix works for me and I've faced no similar issues since.

**Comment by Permanently** [comment]
Also put in a complaint last Friday afternoon, although I didn't realise an update came out at the time. Expecting a callback tomorrow, so I'll be referencing @SvartrShell's message as I was just arguing with the rep about "the definition of Android" whilst driving!

**Comment by SvartrShell** [comment]
Update on this. Had a call regarding the complaint. Handler understandably had very little knowledge on the issue, but basically used the fact the apps terms and conditions state your device must run the trademarked Google certified version of Android.

Very little can be done about it.

I closed my accounts and moved elsewhere.

**Comment by Permanently** [comment]
I had a call back this afternoon as well! Didn't have the "Google" thing mentioned to me; I tried putting it into fairly high-level terms and explaining equally how there's different versions of Windows, the same way that there is for Android. She wasn't having any of it. After nearly half an hour of trying to explain that, I gave up and cut my losses. She didn't really seem that bothered about trying to understand anyway.
